datacube_alchemist/worker.py

```diff
@@ -347,6 +347,11 @@ def _determine_output_product(self):
             if self.config.output.properties:
                 for k, v in self.config.output.properties.items():
                     dataset_assembler.properties[k] = v
+            # add dataset and product maturity properties from original dataset rather than output config
+            if "dea:dataset_maturity" in source_doc.properties:
+                dataset_assembler.properties["dea:dataset_maturity"] = source_doc.properties["dea:dataset_maturity"]
+            if "dea:product_maturity" in source_doc.properties:
```

Comment:
Lets leave `product_maturity` out of this, it should be determined by the configuration file, not the input.